projects/cds/src/cms-components/merchandising/carousel/merchandising-carousel/merchandising-carousel.component.ts

```diff
@@ -0,0 +1,51 @@
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { Product } from '@spartacus/core';
+import { CmsComponentData } from '@spartacus/storefront';
+import { Observable, of } from 'rxjs';
+import { distinctUntilChanged, filter, map, switchMap } from 'rxjs/operators';
+import { CmsMerchandisingCarouselComponent } from '../../../../cds-models/cms.model';
+import { StrategyConnector } from './../../../../merchandising/connectors/strategy/strategy.connector';
+
+@Component({
+  selector: 'cx-merchandising-carousel',
+  templateUrl: './merchandising-carousel.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MerchandisingCarouselComponent {
+  private componentData$: Observable<
+    CmsMerchandisingCarouselComponent
+  > = this.componentData.data$.pipe(filter(Boolean));
+
+  title$: Observable<string> = this.componentData$.pipe(
+    map(data => data.title)
+  );
+
+  size$: Observable<string> = this.componentData$.pipe(
+    map(data => data.numberToDisplay)
+  );
+
+  backgroundColor$: Observable<string> = this.componentData$.pipe(
+    map(data => data.backgroundColour)
+  );
+
+  textColor$: Observable<string> = this.componentData$.pipe(
+    map(data => data.textColour)
+  );
+
+  items$: Observable<Observable<Product>[]> = this.componentData$.pipe(
+    map(data => data.strategy),
+    distinctUntilChanged(),
+    switchMap(strategyId =>
+      this.strategyConnector.loadProductsForStrategy(strategyId)
+    ),
+    map(merchandisingProducts => merchandisingProducts.products),
+    map(products => products.map(product => of(product)))
```

Comment:
I think this line can be removed. This will flatten the return type (line 35) to just `Observable<Product[]>`

Reply:
This can't be removed. Its something Nikola mentioned also, however, if we remove it and only have an Observable<Product[]> then we get console errors due to the AsyncPipe on the cx-carousel. Also, the OCC product carousel does this it in the same way, which is where we based it off of

Reply:
my mistake (forget this :))